sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -548,9 +548,10 @@ object NullPropagation extends Rule[LogicalPlan] {
           Coalesce(newChildren)
         }
 
-      // If the value expression is NULL then transform the In expression to null literal.
+      // If the value expression is NULL then transform the In expression
+      // or PredicateSubquery expression to null literal.
       case In(Literal(null, _), _) => Literal.create(null, BooleanType)
-      case InSubquery(Seq(Literal(null, _)), _) => Literal.create(null, BooleanType)
+      case PredicateSubquery(Seq(Literal(null, _)), _, _) => Literal.create(null, BooleanType)
```

Comment:
This conversion is not correct as I mentioned in https://github.com/apache/spark/pull/24344. But I still keep it here, and will fix this in another PR for [SPARK-27572](https://issues.apache.org/jira/browse/SPARK-27572).